cmd/query.go

```diff
@@ -679,44 +685,220 @@ $ %s query channel ibc-2 ibctwochannel transfer --height 1205`,
 	return heightFlag(a.Viper, cmd)
 }
 
+// chanExtendedInfo is an intermediate type for holding additional useful
+// channel information regarding IBC hierarchy of clients/conns/chans.
+type chanExtendedInfo struct {
+	clientID             string
+	counterpartyChainID  string
+	counterpartyConnID   string
+	counterpartyClientID string
+}
+
+func printChannelWithExtendedInfo(
+	cmd *cobra.Command,
+	chain *relayer.Chain,
+	channel *chantypes.IdentifiedChannel,
+	extendedInfo *chanExtendedInfo) {
+	s, err := chain.ChainProvider.Sprint(channel)
+	if err != nil {
+		fmt.Fprintf(cmd.ErrOrStderr(), "Failed to marshal channel: %v\n", err)
+		return
+	}
+
+	if extendedInfo == nil || len(channel.ConnectionHops) == 0 {
+		fmt.Fprintln(cmd.OutOrStdout(), s)
+		return
+	}
+
+	asJson := make(map[string]any)
+	if err := json.Unmarshal([]byte(s), &asJson); err != nil {
+		fmt.Fprintln(cmd.OutOrStdout(), s)
+		return
+	}
+
+	asJson["chain_id"] = chain.ChainProvider.ChainId()
+	asJson["client_id"] = extendedInfo.clientID
+	counterparty, ok := asJson["counterparty"].(map[string]any)
+	if ok {
+		counterparty["chain_id"] = extendedInfo.counterpartyChainID
+		counterparty["client_id"] = extendedInfo.counterpartyClientID
+		counterparty["connection_id"] = extendedInfo.counterpartyConnID
+	}
+
+	newJson, err := json.Marshal(asJson)
+	if err != nil {
+		fmt.Fprintln(cmd.OutOrStdout(), s)
+		return
+	}
+
+	fmt.Fprintln(cmd.OutOrStdout(), string(newJson))
+}
+
+const concurrentQueries = 10
+
+func queryChannelsToChain(cmd *cobra.Command, chain *relayer.Chain, dstChain *relayer.Chain) error {
+	ctx := cmd.Context()
+
+	clients, err := chain.ChainProvider.QueryClients(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, client := range clients {
+		tmClient, err := relayer.CastClientStateToTMType(client.ClientState)
```

Comment:
It will silently fail, so it will still be able to print out without the extended information. But yes this can be refactored later on to something like a chain provider method to get a chain ID for a specific client.

Reply:
I went ahead and did the refactor now